rest/route/user.go

```diff
@@ -245,27 +257,33 @@ func (h *userPermissionsDeleteHandler) Run(ctx context.Context) gimlet.Responder
 		}))
 		return gimlet.MakeJSONInternalErrorResponder(errors.New("unable to get roles for user"))
 	}
-	scopesIds := []string{}
-	for _, role := range roles {
-		scopesIds = append(scopesIds, role.Scope)
+	rolesToCheck := []gimlet.Role{}
+	// don't remove basic access, just special access
+	for _, r := range roles {
+		if !utility.StringSliceContains(evergreen.BasicAccessRoles, r.ID) {
+			rolesToCheck = append(rolesToCheck, r)
+		}
+	}
+	if len(rolesToCheck) == 0 {
+		gimlet.NewJSONResponse(struct{}{})
```

Comment:
What does this line do? Do we need to specify a return statement here?

Reply:
This just returns an empty successful response; we also do this at 294 at the end of the function but in this case if there are no roles to check we can skip the next query. We should probably also do this in the update function.